test/cql-pytest/test_shedding.py

```diff
@@ -31,12 +32,47 @@ def table1(cql, test_keyspace):
 # 5. Hence, a 30MiB request will be estimated to take around 60MiB RAM,
 #    which is enough to trigger shedding.
 # See also #8193.
-@pytest.mark.skip(reason="highly depends on configuration")
+#
+# Big request causes the entire connection to be closed (see #8800 for the reasons),
+# this results in NoHostAvailable on the client.
+# We check that there are no unexpected protocol_errors using Scylla Prometheus API.
+# Such errors occurred before, when before closing the connection, the remaining
+# bytes of the current request were not read to the end and were treated as
+# the beginning of the next request.
+#
+# Have no idea why, but on CI the actual memory limit for CQL requests is ~100MiB,
```

Comment:
I checked the logic for calculating the limit (it's outlined in this comment above):
```
auto mem_estimate = f.length * 2 + 8000

// 30mb * 2 + 8000 > 50mb - should trigger shedding, and locally it does, but not on the CI
// I dumped the counters (their description includes the actual value of _max_request_size), 
// and it turned out to be 100mb, instead of 50mb for some reason((
if (mem_estimate > _server._max_request_size) {
...
}
==>
// 50mb
cql_server_config.max_request_size <== _mem_limiter.local().total_memory();

// 500mb / 10 -> 50 mb
total_memory <== available_memory / 10

// from SCYLLA_CMDLINE_OPTIONS, --smp 2, -m 1G --> available_memory := 500mb
available_memory <==memory::stats().total_memory() // this is for logical core
```